### PR TITLE
Jbzdak/pruner

### PR DESCRIPTION
**Pull request by jbzdak**
Authors concerns:
- Would suggest to add a task to do some hardening on this intance, like: nodefault ssh, I have no idea what more. 
- This might be split to public/private repository pair, but I doubt it's worth it.

### CODE REVIEW

**Comment by bradenmacdonald** [comment]
I expect all OpenCraft repositories to have a clear README and LICENSE. Here, the README says how to use this but not what it is or why someone would want it. This also needs a LICENSE file.

And I imagine you've already been over this a thousand times in the previous reviews, but why do we have (encrypted) private vars mixed in with code in the same repo? Shouldn't those be separated? And doesn't that make this not re-usable?

**Comment by jbzdak** [comment]
@bradenmacdonald 
1. As noted in the author's concerns: This is a really small piece of actual code, the code part less than 60 lines of yaml, we could split this into two repositories, but then we'd have to keep these repositories in sync, for example using submodules --- which has it's quirks. I'm not sure it's worth the effort. Not that I'm not willing to spend extra work to make world a better place, I'm just pointing out it could be not worth the effort. But if you think it's worth spltting, I'll do this. 
2. I didn't put the license in other private repositories, and it looks like we didn't have a habit of putting license in private repos. I would have added the license to the public version of this repository. We don't want others to give right to copy our encrypted credentials, do we? 
3. Fair point about the readme, I should definetely add some information on what does it do.

**Comment by bradenmacdonald** [comment]
@jbzdak Sorry, I totally didn't notice that this is a private repository. The point about the README still stands but the rest you can ignore.

**Comment by jbzdak** [comment]
@bradenmacdonald  No harm done. Now I see, why you were irritated so much --- I'm aware that pushing (encrypted, but nevertheless) private data to public repository without license and readme should be avoided :)

**Comment by bradenmacdonald** [comment]
@jbzdak I wasn't "irritated", just wanted some changes :) By way of explanation, there is nothing in the name, "deployment-prune-backups" that makes this sound like something that would be private so I didn't even think to check.

**Comment by jbzdak** [comment]
@bradenmacdonald Never mind then. Maybe I should change the name of this repository to something more secretive ;),

**Comment by antoviaque** [comment]
@jbzdak Actually, most of the content of this repo should be on a public repository - that's exactly the same reasoning as the other repositories, I'm not sure why you put everything on the private repo here. Can you change this before I get around to review and test, and check if there are other patterns we discussed in the review previously that this breaks?

**Comment by jbzdak** [comment]
@antoviaque To be 100% clear, if I publish `public.yml` anyone on the internet will be able to: 
1. Enumerate approximate number of services we host 
2. Probably guess something about the nature of services we host (from the fact that we have backup container named: e.g. `stgallen-edxapp`) 
3. Know how often we do backups. 

Should I share these variables?

**Comment by antoviaque** [comment]
@jbzdak Yes, that seem fine to me.

Cc @bradenmacdonald

**Comment by antoviaque** [comment]
@jbzdak The split hasn't been done here yet? This is blocking to be able to test this.

**Comment by jbzdak** [comment]
@antoviaque This is now splitted, see: https://github.com/open-craft/deployment-secrets-prune-backups/pull/1

File: backup-pruner.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    owner: "root"
+    group: "root"
+    mode: "700"
+    state: directory
+  no_log: true
+  with_items: "{{ keys_to_upload }}"
+
+# NOTE: Pip doesn't work with tarsnapper
+- name: install tarsnapper
+  command: easy_install "{{ TARSNAPPER_VERSION }}"
```
Comment (antoviaque) [inline backup-pruner.yml:38]: Why do you need to install tarsnapper manually, when the `pmbauer.tarsnap` role from https://github.com/open-craft/deployment-deploy-prune-backups/pull/1/files#diff-976d348c79e886d511b3d19dc93fa69dR8 should already allow to install it?

File: host_vars/pruner/public.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+
+FORWARD_MAIL_RELAY_MAIL_TO: "ops@example.com"
+FORWARD_MAIL_SMTP_FROM: "{{ FORWARD_MAIL_RELAY_MAIL_TO }}"
+
+TARSNAPPER_CONF: /etc/tarsnapper.yml
+
+# Can be changed to normal tarsnapper once we merge this branch
+# and code winds up on PIP
+TARSNAPPER_VERSION: git+https://github.com/open-craft/tarsnapper@jbzdak/tarsnapper-extensions
```
Comment (antoviaque) [inline host_vars/pruner/public.yml:9]: I don't see any upstream PR for this change (and also no tests?):
- https://github.com/open-craft/tarsnapper/commit/7f2a7ee17fe5451af580d7aeab7719cb58d8a19f
- https://github.com/miracle2k/tarsnapper/pulls

File: host_vars/pruner/public.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      target: dalite-swift-backup-$date
+      keyfile: "{{ DALITE_SWIFT_PATH }}"
+      cachedir: "{{ DALITE_SWIFT_CACHE }}"
+      deltas: 1h 1d 7d 30d 360d 18000d
+    mysql:
+      sources: []
+      target: mysql.opencraft.com-$date
+      keyfile: "{{ MYSQL_BACKUP_PATH }}"
+      cachedir: "{{ MYSQL_BACKUP_CACHE }}"
+      deltas: 1h 1d 7d 30d 360d 18000d
```
Comment (antoviaque) [inline host_vars/pruner/public.yml:24]: These should be in the private repo, with some sample values here

File: templates/prune-job.sh [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+
+/usr/local/bin/tarsnap --fsck --keyfile "{{ item.keypath }}" \
+--cachedir "{{ item.cache }}" && \
+tarsnapper --config /etc/tarsnapper.yml  expire dalite-logs "{{ item.name }}"
```
Comment (antoviaque) [inline templates/prune-job.sh]: why is `dalite-logs` referenced here? It should already be handled by `item.name` no?

File: host_vars/pruner/public.yml (hunk 2 above)
Comment (jbzdak) [inline host_vars/pruner/public.yml:9]: This is now addressed: https://github.com/miracle2k/tarsnapper/pull/40

File: backup-pruner.yml (hunk 1 above)
Comment (jbzdak) [inline backup-pruner.yml:38]: In `pmbauer.tarsnap` tarsnap version is not configurable, plus this role does bunch of things we don't really need --- like running backups in cron. It's more robust to just install tarsnapper here manually that adding bunch of swiches to `pmbauer.tarsnap` IMO.

File: host_vars/pruner/public.yml (hunk 3 above)
Comment (jbzdak) [inline host_vars/pruner/public.yml:24]: Addressed in 65cd051621beae425e45f590c98438a8bf2ca41f

File: templates/prune-job.sh (hunk 4 above)
Comment (jbzdak) [inline templates/prune-job.sh]: Addressed in 65cd051621beae425e45f590c98438a8bf2ca41f . 

It was just copy-paste error. When I checked it worked, as the job I was checking was pruned.

File: backup-pruner.yml (hunk 1 above)
Comment (antoviaque) [inline backup-pruner.yml:38]: OK

**Comment by antoviaque** [comment]
:+1:

**Comment by jbzdak** [comment]
Oh damn. This was not merged.

**Pull request merged**